frontend/src/components/elements/Favicon/Favicon.tsx

```diff
@@ -0,0 +1,99 @@
+/**
+ * @license
+ * Copyright 2018-2020 Streamlit Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { Component, ReactNode } from "react"
+import ReactDOM from "react-dom"
+import { Map as ImmutableMap } from "immutable"
+import { getImageURI } from "../ImageList/ImageList"
+import nodeEmoji from "node-emoji"
+
+export interface Props {
+  element: ImmutableMap<string, any>
+}
+
+/**
+ * Hidden element that overwrites the page's favicon with the provided image
+ */
+export class Favicon extends Component<Props> {
+  emoji = ""
```

Comment:
I think it's better when access modifiers are transparent. Thoughts?

Reply:
I actually switched to a functional implementation to get the component to re-render correctly -- these are now instances of useState